### ISSUE TITLE
Ancient Petrified Leaf Quest and stave-of-the-ancients no scripts 1.12.1

### ISSUE THREAD

**Issue by e46ad** [scriptdev2/scriptdev2-classic#37]
I see this in my client SD2 File,

No script found for ScriptName 'at_ancient_leaf'.
No script found for ScriptName 'go_fathom_stone'.

Quest http://www.wowhead.com/item=18703/ancient-petrified-leaf Is bugged i can't find the npc below i did this quest in vanilla myself and something is not correct with it.

http://www.wowhead.com/npc=14524/vartrus-the-ancient

I don't see vartrus-the-ancient in game i have the Leaf from molten core, I've been to the place he is suppose to spawn when on the quest but his not there, I had to spawn him with a GM command to able to see him and advance to the next stage of the quest he give's ( A DB error will post on Cmangos, I know you can't fix that)

http://www.wowhead.com/quest=7636/stave-of-the-ancients

I see each Npc in game but they don't have the required scripts, I can talk to them but they will just Say good day My Character Name, And some of them have custom text but no trigger to start the encounter and kill the creature for the Head's that are a required item for the Bow and Staff,

Latest Git pull of Scriptdev2-Classic with Cmangos latest DB and Core 1.12.1,

there my be an available fix but the website is still offline, So posting here so i can get conformation of a fix or if it's yet to be added or i've missed an update or something,

I tired this on a admin/gm account and a normal players account, each time same results,

Thanks for reading.

**Comment by Rushor** [comment]
https://github.com/scriptdev2/scriptdev2/blob/master/sql/updates/r3070_mangos.sql
https://github.com/scriptdev2/scriptdev2/blob/master/scripts/world/areatrigger_scripts.cpp#L369

**Comment by e46ad** [comment]
Hi thanks for the reply what do i do with that .cpp file i know what to do with that SQL file,

Whats the second link do is it a new script for them ? were do i place the file or do i just copy it and save it as a cpp or an SQL and extract it to my mangos db, Or do i add it to my SD2 Folder ?

**Comment by Rushor** [comment]
the c++ changes are neededin order to trigger the summon of the npcs. simply store the stuff in the c++ files and recompile the core.

add the db changes to the fitting db and restart the server after the compiling is done :)

**Comment by e46ad** [comment]
Hey thanks for the fast replies i just have a few questions if you don't mind helping me please,

What do i do with the second link you sent do i just copy everything in the note and then paste it into notepad and save as a .cpp file ?

Second what folder do i place it in to re compile is it my C:\Users\AD\mangos-classic\src\bindings\scriptdev2-classic and then place it into the scripts folder and re compile SD2 ?


Or is that the wrong place to add the file sorry just new to this and learning what do to fix things,

Thanks ever so much for the replies

**Comment by Rushor** [comment]
jep the last one, open up the areatrigger_script.cpp and add all changes into the file and save it. after that simply recompile it, as you compiled the core on the first time

**Comment by e46ad** [comment]
Great cheers pal, Last question is Should i re compile my whole mangos core? or just the scriptdev2 folder ? in Src/bindings/scriptdev2 or just compile using the Win MangosdVC120.sln ? 

Cheers for the help much appreciated.

**Comment by Rushor** [comment]
hm i think it was just the scriptdev2

and yeah simply run the sql in your world db

**Comment by e46ad** [comment]
Hey, eddied the .cpp file in SD2 and the SQL file, All Npc's are there now but they still don't have the option to transform them into a dragon creature to kill them to loot the heads, All 4 Npc's still don't have the scripts still :( did all you said and compile has zero errrors for SD2, 

Could you upload a screenshot of your NPC's please, and see if they have the correct scripts active and npc text thanks for the help 

![art](https://cloud.githubusercontent.com/assets/11399502/6656966/68d4c662-cb33-11e4-8412-7ab184a84d02.jpg)
![franklin](https://cloud.githubusercontent.com/assets/11399502/6656967/6d9d5452-cb33-11e4-8e91-6e975d3f5fb8.jpg)
![nelson the nice](https://cloud.githubusercontent.com/assets/11399502/6656968/710db2d0-cb33-11e4-8e62-3f81c6030275.jpg)
![simone](https://cloud.githubusercontent.com/assets/11399502/6656969/74e44054-cb33-11e4-81c5-549e6754132a.jpg)

**Comment by Rushor** [comment]
ohh damn i just saw at_ancient_leaf and thought that you want to know how to add the script for it.
but the script for at_ancient_leaf does not actually fix the quest http://www.wowhead.com/quest=7636/stave-of-the-ancients. so sry the event for these npcs must be fixed first from someone ^^

**Comment by e46ad** [comment]
Oh okay so it's not fixed right now ? I fixed the error in my SD2 logs now but i get new ones should i worry about these ?

2015-03-15 17:12:11 Script registering but ScriptName at_aldurthar_gate is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_coilfang_waterfall is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_legion_teleporter is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_spearborn_encampment is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_warsong_farms is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_waygate is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_stormwright_shelf is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName at_hot_on_the_trail is not assigned in database. Script will not be used.
2015-03-15 17:12:11 Script registering but ScriptName boss_arugal is not assigned in database. Script will not be used.

Thats after i used the new updates from your .cpp file should i be worried about these or not ?

**Comment by Rushor** [comment]
it say that you are missing some db entries, are you shure, that you applied all db updates in you world db?

**Comment by e46ad** [comment]
Yeah i applied every single DB update i could from the DB and Mangos sql updates and SD2 Updates, Don't know why it's just started to report that only when i look in my SD2 log's i see it tho it doesn't seam to stop the server working right now. 

Any ideas?
